### ISSUE TITLE
Training Process didn't progress.

### ISSUE THREAD

**Issue by jageshmaharjan** [mozilla/DeepSpeech#1371]
For support and discussions, please use our [Discourse forums](https://discourse.mozilla.org/c/deep-speech).

If you've found a bug, or have a feature request, then please create an issue with the following information:

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: No Custom Code
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: pip install tensorflow-gpu
- **TensorFlow version (use command below)**: 
- **Python version**: 1.7.0
- **Bazel version (if compiling from source)**: 0.12.0
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: CUDA 9.0 and cuDNN 7.0.3
- **GPU model and memory**: NVIDIA Tesla M60 
- **Exact command to reproduce**:
 `python DeepSpeech.py --train_files cv_data/cv-valid-train.csv --dev_files cv_data/cv-valid-dev.csv --test_files cv_data/cv-valid-test.csv --validation_step=1`

You can obtain the TensorFlow version with

```bash
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
```
I was trying to train with the "Common Voice" dataset downloaded and parsed with `bin/import_cv.py cv_data/`.
The training process started with no errors, but its in the same state as shown in the trace below, with not progress. I was wondering if the process is in progress or stuckup with some unknown event. Its been more than 6 hours.

`/home/maybe/anaconda3/envs/asr/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
WARNING:tensorflow:From /home/maybe/anaconda3/envs/asr/lib/python3.6/site-packages/tensorflow/contrib/learn/python/learn/datasets/base.py:198: retry (from tensorflow.contrib.learn.python.learn.datasets.base) is deprecated and will be removed in a future version.
Instructions for updating:
Use the retry module or similar alternatives.
I STARTING Optimization
`

**Comment by jageshmaharjan** [comment]
yup, the GPU is being utilized that process. I checked that earlier too,

`+-----------------------------------------------------------------------------+
| NVIDIA-SMI 390.30                 Driver Version: 390.30                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla M60           Off  | 00007815:00:00.0 Off |                  Off |
| N/A   59C    P0    83W / 150W |   7765MiB /  8129MiB |     94%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla M60           Off  | 00009062:00:00.0 Off |                  Off |
| N/A   60C    P0    84W / 150W |   7765MiB /  8129MiB |     79%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla M60           Off  | 0000A7D4:00:00.0 Off |                  Off |
| N/A   65C    P0    88W / 150W |   7765MiB /  8129MiB |     89%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla M60           Off  | 0000C057:00:00.0 Off |                  Off |
| N/A   53C    P0    87W / 150W |   7765MiB /  8129MiB |     91%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     68580      C   python                                      7746MiB |
|    1     68580      C   python                                      7746MiB |
|    2     68580      C   python                                      7746MiB |
|    3     68580      C   python                                      7746MiB |
+-----------------------------------------------------------------------------+`

**Comment by kdavis-mozilla** [comment]
Then I'd almost guess that things are going normally.

I've never used the NVIDIA Tesla M60 for Deep Speech and Common Voice. So, I'm not sure how long an epoch takes.

However, looking at the command you are running
```bash
python DeepSpeech.py --train_files cv_data/cv-valid-train.csv --dev_files cv_data/cv-valid-dev.csv --test_files cv_data/cv-valid-test.csv --validation_step=1
```
and the default parameters [DeepSpeech.py#L82](https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L82)
```python
...
tf.app.flags.DEFINE_integer ('train_batch_size', 1,           'number of elements in a training batch')
tf.app.flags.DEFINE_integer ('dev_batch_size',   1,           'number of elements in a validation batch')
tf.app.flags.DEFINE_integer ('test_batch_size', 1, 'number of elements in a test batch')
...
```
I'd guess your training batch size is `1` which is relatively small and will make the run slow. I'd guess you might try increasing the training batch size to `4` or  `5` say, for example
```bash
python DeepSpeech.py --train_batch_size 4 --train_files cv_data/cv-valid-train.csv --dev_files cv_data/cv-valid-dev.csv --test_files cv_data/cv-valid-test.csv --validation_step=1
```
I'm not sure where the out of memory (OOM) error will kick in with your GPU and Common Voice so this might be too large or not large enough. However, this increase in should get you about a factor of `4` or `5` speedup.

**Comment by jageshmaharjan** [comment]
Thanks @kdavis-mozilla , I'll wait for a bit, could be running as well because all resources are utilized at this moment. Incase, if it doesn't show any further info, I'll change the batch size.

**Comment by jageshmaharjan** [comment]
wow, noticed that the resources (CPU, GPU, and Memory) utilization is super Huge. That's might be the reason. Not too sure yet.

**Comment by jageshmaharjan** [comment]
Yup, the issue had been solved, sorry for not closing on time. 
Thanks for closing the issue.

**Comment by AlexPeng19** [comment]
so it is normal for no message printed? could i know how did you resolve it, change the batch size or what else. i have encounterd the same problem. but i have not gpu, no sure it matters or not. only noticed the cpu load is high too. Looking forward your response. @jageshmaharjan @kdavis-mozilla

**Comment by kdavis-mozilla** [comment]
@AlexPeng19 Training is very slow without a GPU, and it may look as if no progress is being made.

**Comment by AlexPeng19** [comment]
that's my command: 
python3 ./DeepSpeech.py --train_files ../Downloads/cv-valid-train.csv --dev_files ../Downloads/cv-valid-dev.csv --test_files ../Downloads/cv-valid-test.csv --decoder_library_path ./libctc_decoder_with_kenlm.so --validation_step=1 >run.log
currently there is no message printed, i have been run it for two days. should i make extra configuration for the log print.
@kdavis-mozilla

**Comment by kdavis-mozilla** [comment]
@AlexPeng19 Dumb question, but I have to ask: When you say "there is no message printed" do you mean there is no message printed to the console or do you mean that the file `run.log` is empty?

**Comment by AlexPeng19** [comment]
neither of them have message.

**Comment by AlexPeng19** [comment]
i got the message: I STARTING Optimization, i think the cause is too slow. thanks for patient answers. 
on the other hand, while i run the kaldi, the log is more friendly.

**Comment by AlexPeng19** [comment]
@kdavis-mozilla

**Comment by kdavis-mozilla** [comment]
@AlexPeng19 Can we move this to [discourse](https://discourse.mozilla.org/c/deep-speech) as there seems to be no bug or feature request just questions on how the system trains only using a CPU.
